ml/misc/queue.py

```diff
@@ -0,0 +1,81 @@
+"""
+ml/misc/queue
+"""
+
+
+class Queue:
+    """
+    Queue modulo using list
+    """
+
+    def __init__(self, init_list=[]):
+        """
+        initiator of Queue modulo
+        """
+        if not isinstance(init_list, list):
+            self.head = []
```

Comment:
It is really bad to name it `self.head` which is really the data for whole queue. Either `self.data` or `self.list` sounds better.